File: backend/app/schemas/explain.py
from typing import Literal

from pydantic import BaseModel

from app.schemas.program_api import Advisory


class TemplateExplanationOut(BaseModel):
    template_id: int
    slug: str
    name: str
    fit_pct: int
    factors: dict[str, float]
    tier: Literal["best", "strong", "possible"]
    advisories: list[Advisory] = []


class LedgerContributionOut(BaseModel):
    group: str
    effective_sets: float


class SlotExplanationOut(BaseModel):
    workout_exercise_id: int
    exercise_id: int
    exercise_name: str
    factors: dict[str, float]
    score: float
    ledger_contributions: list[LedgerContributionOut]
    safety_note: str | None

File: frontend/src/types/explain.ts
import type { Advisory } from '@/types/program';

export interface TemplateExplanation {
  template_id: number;
  slug: string;
  name: string;
  fit_pct: number;
  factors: Record<string, number>;
  tier: 'best' | 'strong' | 'possible';
  advisories: Advisory[];
}

export interface LedgerContribution {
  group: string;
  effective_sets: number;
}

export interface SlotExplanation {
  workout_exercise_id: number;
  exercise_id: number;
  exercise_name: string;
  factors: Record<string, number>;
  score: number;
  ledger_contributions: LedgerContribution[];
  safety_note: string | null;
}
